numpyro/mcmc.py

```diff
@@ -80,7 +80,8 @@ def init_kernel(init_samples,
                              wa_state.step_size, wa_state.inverse_mass_matrix, rng_hmc)
 
         if run_warmup:
-            hmc_state, _ = fori_loop(0, num_warmup_steps, warmup_update, (hmc_state, wa_state))
+            hmc_state, _ = jit(fori_loop, static_argnums=(2,))(0, num_warmup_steps, warmup_update,
+                                                               (hmc_state, wa_state))
```

Comment:
hmm, does this really help? I thought we just run it 1 time so jitting is not useful.

Reply:
Both of these help. Yes, please test. My hypothesis is that it helps because it compiles `sample_kernel` essentially which is reused across time steps.

Reply:
Yup, I can confirm that it helps. :D